src/ray/common/memory_monitor.cc

```diff
@@ -111,41 +111,78 @@ std::tuple<int64_t, int64_t> MemoryMonitor::GetMemoryBytes() {
   return std::tuple(system_used_bytes, system_total_bytes);
 }
 
-int64_t MemoryMonitor::GetCGroupV1MemoryUsedBytes(const char *path) {
+int64_t MemoryMonitor::GetCGroupV1MemoryUsedBytes(const char *stat_path,
+                                                  const char *usage_path) {
+  // Used memory calculation from cgroup info:
+  //  used_memory = `memory.usage_in_bytes` - `memory.stat.total_cache - memory.stat.shmem`
+  // This value is consistent with values `MemTotal` `MemAvailable` `MemFree` in `/proc/meminfo`
+  //  and they have relationship of:
+  //  - `memory.usage_in_bytes` == `MemTotal` - `MemFree`
+  //  - `memory.stat.total_cache` == `MemAvailable` - `MemFree`
+  //  - `memory.stat.total_cache` == OS_managed_cache_and_buffer + `/dev/shm_usage`
+  //
+  // Explanation: What's this part `OS_managed_cache_and_buffer` memory for and why
+  //   we should treat this part memory as "Not-in-used" ?
+  // Linux OS tries to fully use the whole physical memory size, so that in many
+  // cases we can observe that the system has very little `MemFree` size,
+  // but at the same time system might have a large `MemAvailable` size,
+  // then this part `MemAvailable - MemFree` size is borrowed by Linux OS
+  // to cache pages / buffers , BUT, once user process requests to allocate memory,
+  // and there is no sufficient free memory, OS will evict cache data out of this
+  // part memory and allocate them to user proces.
+  //
```

Comment:
NOTE: 

Current master Ray code calculation counts "OS_managed_cache_and_buffer" as "in-used" memory, which causes the issue that it heavily underestimate available memory,
and it leads to Ray memory monitor misleadingly triggered to kill Ray tasks while there are lots of available memory.

This is a very big issue.

Because for some IO-heavy tasks, it can quickly fill up physical memory with "OS_managed_cache_and_buffer", then Ray monitor will misleadingly think OS has very few available memory and then kill Ray tasks.